cirq/contrib/acquaintance/permutation.py

```diff
@@ -199,3 +243,26 @@ def __init__(self):
                  not isinstance(op.gate, SwapPermutationGate)]))
 
 expand_permutation_gates = ExpandPermutationGates()
+
+
+def return_to_initial_mapping(circuit: circuits.Circuit,
+                              swap_gate: ops.Gate = ops.SWAP) -> None:
+    qubits = sorted(circuit.all_qubits())
+    n_qubits = len(qubits)
+
+    mapping = {q: i for i, q in enumerate(qubits)}
+    update_mapping(mapping, circuit.all_operations())
+
+    permutation = {i: mapping[q] for i, q in enumerate(qubits)}
+    returning_permutation_op = LinearPermutationGate(n_qubits, permutation,
+                                                     swap_gate)(*qubits)
+    circuit.append(returning_permutation_op)
+
+
+def uses_consistent_swap_gate(circuit: circuits.Circuit,
+                              swap_gate: ops.Gate) -> bool:
+    for op in circuit.all_operations():
+        gate = ops.op_gate_of_type(op, Type[PermutationGate])
```

Comment:
Any thoughts on how to do the typing here? Mypy doesn't like [instantiating a typevar with an abstract class](https://github.com/python/mypy/issues/5374).

Reply:
I reverted so the tests pass, but if you see a way to use `op_gate_of_type` that mypy finds acceptable, let me know.